spark/src/main/scala/io/delta/tables/DeltaColumnBuilder.scala

```diff
@@ -96,6 +99,69 @@ class DeltaColumnBuilder private[tables](
     this
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Specify a column as an identity column that is always generated by the system, i.e., does not
+   * allow user specified values using a start of 1 and a step of 1.
+   *
+   * @since 3.2.0
+   */
+  @Evolving
+  def generatedAlwaysAsIdentity(): DeltaColumnBuilder = {
+    generatedAlwaysAsIdentity(start = 1, step = 1)
+  }
+
+  /**
+   * :: Evolving ::
+   *
+   * Specify a column as an identity column that is always generated by the system, i.e., does not
+   * allow user specified values.
+   *
+   * @param start the start value of the identity column
+   * @param step the increment step of the identity column
+   * @since 3.2.0
```

Comment:
Wow, glad to see this PR so quickly. Are you planning to release it in 3.2.0?
Thanks.

Reply:
It's a stretch goal at the moment, since the release for Delta 3.2 is quite soon. I should know by end of week whether I can realistically make 3.2. I plan to leave this PR up for at least one week anyways. :)

It's safe to start merging the code even if we can only make 4.0 since the DeltaSQLConf guards creating identity columns.

Reply:
For context, this PR is on the critical path for identity columns implementation in Delta 3.2. We need an interface to specify identity columns. The SQL syntax can only be introduced in the next Spark release (Spark 4), which means we can only have it for Delta 4.0 earliest.

Reply:
Got it. Just for my knowledge, the required changes in Spark 4 are only to support SQL syntax, meaning Scala could be supported?

Reply:
That's part of what I'm assessing this week in my prototype. To put the UDF for generating identity column values in  only the Delta repo requires adding workarounds so that it works on Spark 3.5. Once I have enough confidence about the idea, I'll post a PR.

Everything else doesn't seem to require Spark changes.

Reply:
I'm not going to make Delta 3.2. I've discussed with @tdas offline and we're aiming for the next Delta version, Delta 3.3. The API doc comment in this PR has been adjusted accordingly.